packages/next/client/index.js

```diff
@@ -323,14 +329,70 @@ function renderReactElement(reactEl, domEl) {
 
   if (onPerfEntry && ST) {
     try {
-      const observer = new PerformanceObserver(list => {
-        list.getEntries().forEach(onPerfEntry)
-      })
-      // Start observing paint entry types.
-      observer.observe({
-        type: 'paint',
-        buffered: true,
-      })
+      function getSupportedTypes(types) {
+        if (PerformanceObserver && PerformanceObserver.supportedEntryTypes) {
```

Comment:
`if (PerformanceObserver)` will error if `PerformanceObserver` is not supported. I think you want to check `if (self.PerformanceObserver)` (or something like that).

While I technically it's fine because this is in a try/catch block, I think the code is clearer if you're not relying on that to do feature detection (or if you do, put a comment explaining why).

Reply:
done